prototype_source/lite_interpreter.rst

```diff
@@ -0,0 +1,229 @@
+
+.. code:: java
+
+    // mModule = Module.load(MainActivity.assetFilePath(getApplicationContext(), "deeplabv3_scripted.pt"));
+    mModule = LiteModuleLoader.load(MainActivity.assetFilePath(getApplicationContext(), "deeplabv3_scripted.ptl"));
+
+5. **Test app**: Build and run the `ImageSegmentation` app in Android Studio
+
+iOS
+-------------------
+Get ImageSegmentation demo app in iOS: https://github.com/pytorch/ios-demo-app/tree/master/ImageSegmentation
+
+1. **Prepare model**: Same as Android.
+
+2. **Build libtorch lite for android**:
+
+.. code-block:: bash
+
+   BUILD_PYTORCH_MOBILE=1 IOS_PLATFORM=SIMULATOR BUILD_LITE_INTERPRETER=1 ./scripts/build_ios.sh
+
+
+3. **Remove Cocoapods from the project**:
+
+.. code-block:: bash
+
+   pod deintegrate
+
+4. **Link ImageSegmentation demo app with the custom built library**:
+Open your project in XCode, go to your project Target’s **Build Phases - Link Binaries With Libraries**, click the **+** sign and add all the library files located in `build_ios/install/lib`. Navigate to the project **Build Settings**, set the value **Header Search Paths** to `build_ios/install/include` and **Library Search Paths** to `build_ios/install/lib`.
+In the build settings, search for **other linker flags**. Add a custom linker flag below
+```
+-all_load
+```
+Finally, disable bitcode for your target by selecting the Build Settings, searching for Enable Bitcode, and set the value to **No**.
+
+5. **Update library and api**
+
+  5.1 Update ``TorchModule.mm``: To use the custom built libraries the project, replace `#import <LibTorch/LibTorch.h>` (in ``TorchModule.mm``) which is needed when using LibTorch via Cocoapods with the code below:
+
+.. code-block:: swift
+
+    //#import <LibTorch/LibTorch.h>
+    #include "ATen/ATen.h"
+    #include "caffe2/core/timer.h"
+    #include "caffe2/utils/string_utils.h"
+    #include "torch/csrc/autograd/grad_mode.h"
+    #include "torch/script.h"
+    #include <torch/csrc/jit/mobile/function.h>
+    #include <torch/csrc/jit/mobile/import.h>
+    #include <torch/csrc/jit/mobile/interpreter.h>
+    #include <torch/csrc/jit/mobile/module.h>
+    #include <torch/csrc/jit/mobile/observer.h>
+
+.. code-block:: swift
+
+    @implementation TorchModule {
+    @protected
+    // torch::jit::script::Module _impl;
+     torch::jit::mobile::Module _impl;
+    }
+
+    - (nullable instancetype)initWithFileAtPath:(NSString*)filePath {
+      self = [super init];
+      if (self) {
+          try {
+              _impl = torch::jit::_load_for_mobile(filePath.UTF8String);
+             //  _impl = torch::jit::load(filePath.UTF8String);
+             //  _impl.eval();
+            } catch (const std::exception& exception) {
+                NSLog(@"%s", exception.what());
+                return nil;
+            }
+        }
+        return self;
+    }
+
+
+5.2 Update ``ViewController.swift``
+
+.. code-block:: swift
+
+    //  if let filePath = Bundle.main.path(forResource:
```

Comment:
I think if we save the model as `.pt` rather than `.ptl`, the swift code can still be reused.

Reply:
Yeah it will help reducing the code change in the app. But I'm a bit concerned about mixing these two different model types and thought it's harder to explain it to users if they use the wrong model type in jit by accident.

model.ptl -> lite interpreter is ok
model.pt -> lite interpreter will throw error

model.ptl -> jit is ok
model.pt -> jit is ok